### PR TITLE
Sb cherry pick pr 6466

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Cherry pick of PR [6466](https://github.com/mapbox/mapbox-navigation-android/pull/6466).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,29 @@ Mapbox welcomes participation and contributions from everyone.
...
+          .build()
+  }
+  ```
+- Added guarantees that route progress with `RouteProgress#currentState == OFF_ROUTE` arrives earlier than `NavigationRerouteController#reroute` is called. [#6764](https://github.com/mapbox/mapbox-navigation-android/pull/6764)
+- Fixed a rare `java.lang.NullPointerException: Attempt to read from field 'SpeechAnnouncement PlayCallback.announcement' on a null object reference` crash in `PlayCallback.getAnnouncement`. [#6760](https://github.com/mapbox/mapbox-navigation-android/pull/6760)
+- Fixed standalone `MapboxManeuverView` appearance when the app also integrates Drop-In UI. [#6774](https://github.com/mapbox/mapbox-navigation-android/pull/6774)
+- Introduced `NavigationViewListener.onSpeedInfoClicked` that would be triggered when `MapboxSpeedInfoView` is clicked upon. [#6770](https://github.com/mapbox/mapbox-navigation-android/pull/6770)
+- Each newly instantiated MapboxRouteArrowView class will initialize the layers with the provided options on the first render call. Previously this would only be done if the layers hadn't already been initialized.  [#6466](https://github.com/mapbox/mapbox-navigation-android/pull/6466)
```
Comment (Zayankovsky) [inline CHANGELOG.md:32]: Just like in https://github.com/mapbox/mapbox-navigation-android/pull/6815#discussion_r1066161453 you have to clean up the changelog. That always happens with cherry-picks, but that won't be an issue once we adopt the new changelog strategy by @SevaZhukov.

Comment (cafesilencio) [reply CHANGELOG.md:32]: This will lead to a conflict and a rebase will create a mess just like the last one.  Here we go...

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,8 +5,7 @@ Mapbox welcomes participation and contributions from everyone.
 ## Unreleased
 #### Features
 #### Bug fixes and improvements
-- Fixed an issue with `NavigationView` that caused info panel to shrink in landscape mode with a full screen theme. [#6811](https://github.com/mapbox/mapbox-navigation-android/pull/6811)
-- Fixed standalone `MapboxManeuverView` appearance when the app also integrates Drop-In UI. [#6810](https://github.com/mapbox/mapbox-navigation-android/pull/6810)
```
Comment (Zayankovsky) [inline CHANGELOG.md:9]: These 2 lines need to be restored, those are the changes that I cherry-picked.

**Review by Zayankovsky — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (cafesilencio) [reply CHANGELOG.md]: restored

File: CHANGELOG.md (hunk 1 above)
Comment (cafesilencio) [reply CHANGELOG.md]: Fortunately no conflict this time.

**Pull request merged**